tests/release/releaseLib.go

```diff
@@ -77,6 +81,115 @@ func CreateComponent(devFw framework.Framework, devNamespace, appName, compName,
 	return component
 }
 
+// CreateComponentWithNewBranch will create a new branch, then create the component based on the new branch
+func CreateComponentWithNewBranch(f framework.Framework, testNamespace, applicationName, componentRepoName, componentRepoURL, gitRevision, contextDir, dockerFilePath string, buildPipelineBundle map[string]string) (*appservice.Component, string, string) {
+	var buildPipelineAnnotation map[string]string
+
+	componentName := fmt.Sprintf("%s-%s", "test-component-pac", util.GenerateRandomString(6))
+	testPacBranchName := constants.PaCPullRequestBranchPrefix + componentName
+	componentBaseBranchName := fmt.Sprintf("base-%s", util.GenerateRandomString(6))
+
+	err := f.AsKubeAdmin.CommonController.Github.CreateRef(componentRepoName, "main", gitRevision, componentBaseBranchName)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	if buildPipelineBundle["build.appstudio.openshift.io/pipeline"] != string(`{"name": "fbc-builder", "bundle": "latest"}`) {
+		// deal with some custom pipeline bundle there 
+		buildPipelineAnnotation = build.GetDockerBuildPipelineBundle()
+	} else {
+		buildPipelineAnnotation = constants.DefaultFbcBuilderPipelineBundle
+	}
+
+	componentObj := appservice.ComponentSpec{
+		ComponentName: componentName,
+		Application:   applicationName,
+		Source: appservice.ComponentSource{
+			ComponentSourceUnion: appservice.ComponentSourceUnion{
+				GitSource: &appservice.GitSource{
+					URL:           componentRepoURL,
+					Revision:      componentBaseBranchName,
+					Context:       contextDir,
+					DockerfileURL: dockerFilePath,
+				},
+			},
+		},
+	}
+
+	testComponent, err := f.AsKubeAdmin.HasController.CreateComponent(componentObj, testNamespace, "", "", applicationName, true, utils.MergeMaps(utils.MergeMaps(constants.ComponentPaCRequestAnnotation, constants.ImageControllerAnnotationRequestPublicRepo), buildPipelineAnnotation))
+	Expect(err).NotTo(HaveOccurred())
+
+	return testComponent, testPacBranchName, componentBaseBranchName
+}
+
+func CreatePushSnapshot(devWorkspace, devNamespace, appName, compRepoName, pacBranchName string, pipelineRun *pipeline.PipelineRun, component *appservice.Component) *appservice.Snapshot {
+	var prSHA string
```

Comment:
you could also using the `var{ ...}` here.

Reply:
Updated, thx